pkg/gitmanager/setup.go

```diff
@@ -174,3 +166,53 @@ func (gm *gitManager) initGit(ctx context.Context, payload *core.Payload, oauth
 	}
 	return nil
 }
+
+func (gm *gitManager) makeDownloadRequest(ctx context.Context, downloadURL string,
```

Comment:
Please use `requestutils` package instead of defining this new function

Reply:
This is refactor code, I have taken out common code from function `downloadFile`
Agree with your comment , will change it